### PR TITLE
Type checking:  adds more files to pride

### PR DESCRIPTION
**Pull request by samouri**
Followup to https://github.com/ampproject/amphtml/pull/34093, which was a followup to https://github.com/ampproject/amphtml/pull/33957

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -412,16 +412,16 @@ export class Log {
...
    * @return {!Element} The value of shouldBeTrueish.
-   * @template T
    * @closurePrimitive {asserts.matchesReturn}
    */
   assertElement(shouldBeElement, opt_message) {
+    const shouldBeTrueish = shouldBeElement && shouldBeElement.nodeType == 1;
     this.assertType_(
       shouldBeElement,
-      isElement(shouldBeElement),
+      shouldBeTrueish,
```
Comment (rcebulko) [inline src/log.js:424]: Is this revert intentional?

File: build-system/tasks/check-types.js [hunk 2]
```diff
@@ -40,11 +40,41 @@ const PRIDE_FILES_GLOBS = [
   // Core
   'src/core/**/*.js',
 
+  // Polyfills
+  'src/polyfills/abort-controller.js',
```
Comment (rcebulko) [inline build-system/tasks/check-types.js:44]: Are polyfills required to clear something that depends on them? If not, I'd say leave them out for now; I have a branch that's about to be PR'd that moves them all into core

File: src/log.js (hunk 1 above)
Comment (samouri) [reply src/log.js]: Not at all! Must have gotten out of date with main with a bad rebase.
I just ran `git checkout main src/log.js` and reapplied the necessary changes.

**Review by samouri — COMMENTED** [review]
Reverted polyfills, and fixed the bad rebase.

File: build-system/tasks/check-types.js (hunk 2 above)
Comment (samouri) [reply build-system/tasks/check-types.js]: Removed

**Review by rcebulko — APPROVED** [review]

**Review by rsimha — APPROVED** [review]

**Comment by rcebulko** [comment]
Part of https://github.com/ampproject/amphtml/issues/34096

**Pull request merged**